File: typescript/dni_ruc.ts
// Consulta DNI y RUC — mblapirest.com
const token = "<tu_token>";
const base = "https://mblapirest.com/api/v1";

async function consultarDni(numero: string) {
  const respuesta = await fetch(`${base}/dni/${numero}`, {
    headers: { Authorization: `Bearer ${token}` },
  });
  return respuesta.json();
}

async function consultarRuc(numero: string) {
  const respuesta = await fetch(`${base}/ruc/${numero}`, {
    headers: { Authorization: `Bearer ${token}` },
  });
  return respuesta.json();
}

console.log(await consultarDni("12345678"));
console.log(await consultarRuc("20123456789"));

File: dart/dni_ruc.dart
// Consulta DNI y RUC — mblapirest.com
// pubspec.yaml: http: ^1.0.0
import 'package:http/http.dart' as http;

const token = '<tu_token>';
const base = 'https://mblapirest.com/api/v1';

Future<String> consultar(String ruta) async {
  final respuesta = await http.get(
    Uri.parse('$base$ruta'),
    headers: {'Authorization': 'Bearer $token'},
  );
  return respuesta.body;
}

void main() async {
  print(await consultar('/dni/12345678'));
  print(await consultar('/ruc/20123456789'));
}
